src/main/scala/mesosphere/marathon/core/deployment/impl/TaskReplaceActor.scala

```diff
@@ -74,15 +74,25 @@ class TaskReplaceActor(
     deploymentManagerActor ! HealthStatusRequest(pathId)
   }
 
+  // sort instances with instances having oldest tasks first. Instances with no task are at the end.
+  def sortByOldestFirst(instances: mutable.Queue[Instance]): mutable.Queue[Instance] = {
+    val instancesByIncarnation = instances.partition(_.tasksMap.size != 0)
+    instancesByIncarnation._1.sortBy(_.appTask.status.stagedAt) ++ instancesByIncarnation._2
```

Comment:
The instances in the instancesByIncarnation._2 case are kind of unhealthy ones as they don't have task. Maybe we should put them first then put the ordered tasks?